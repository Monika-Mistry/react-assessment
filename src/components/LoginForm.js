import React from 'react';
import {
    Form,
    FormGroup,
    Label,
    Input,
    Col,
    Button,
    FormFeedback
} from 'reactstrap';

export default function LoginForm(props) {
    const isLoggedIn = props.isLoggedIn;
    const user = props.user;
    const isPassword = props.isPassword;
    const isUser = props.isUser;

    return (
        <Form className="container" onSubmit={props.loginUser}>
            <FormGroup row>
                <Label for="username" sm={2}>Username:</Label>
                <Col sm={10}>
                    <Input
                        required
                        type="text"
                        name="username"
                        id="username"
                        placeholder="Enter your username"
                        invalid={!isUser}
                        valid={isLoggedIn}
                    />
                    <FormFeedback invalid>Username is incorrect</FormFeedback>
                </Col>
            </FormGroup>
            <FormGroup row>
                <Label for="password" sm={2}>Password:</Label>
                <Col sm={10}>
                    <Input
                        required
                        type="password"
                        name="password"
                        id="password"
                        placeholder="Enter your password"
                        invalid={!isPassword}
                        valid={isLoggedIn}
                    />
                    <FormFeedback invalid>Password is incorrect</FormFeedback>
                </Col>
            </FormGroup>
            <div className="text-center">
                <Button>Login</Button>
            </div>
            <br />
            {isLoggedIn && <h1 className="text-center text-success">{user} Logged In</h1>}
        </Form>
    )
}