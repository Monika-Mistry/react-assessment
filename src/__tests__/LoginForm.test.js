import React from 'react';
import renderer from 'react-test-renderer';

import LoginForm from '../components/LoginForm';

it('renders a login form', () => {
    const wrapper = renderer.create(<LoginForm />).toJSON();
    expect(wrapper).toMatchSnapshot();

})

it('renders a successful login', () => {
    const wrapper = renderer.create(<LoginForm user="hello" isLoggedIn={true} />).toJSON();
    expect(wrapper).toMatchSnapshot();
})