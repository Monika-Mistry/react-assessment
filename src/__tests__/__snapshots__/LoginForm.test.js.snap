// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a login form 1`] = `
<form
  className="container"
>
  <div
    className="row form-group"
  >
    <label
      className="col-sm-2 col-form-label"
      htmlFor="username"
    >
      Username:
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="username"
        name="username"
        placeholder="Enter your username"
        required={true}
        type="text"
      />
      <div
        className="invalid-feedback"
        invalid={true}
      >
        Username is incorrect
      </div>
    </div>
  </div>
  <div
    className="row form-group"
  >
    <label
      className="col-sm-2 col-form-label"
      htmlFor="password"
    >
      Password:
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Enter your password"
        required={true}
        type="password"
      />
      <div
        className="invalid-feedback"
        invalid={true}
      >
        Password is incorrect
      </div>
    </div>
  </div>
  <div
    className="text-center"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
  <br />
</form>
`;

exports[`renders a successful login 1`] = `
<form
  className="container"
>
  <div
    className="row form-group"
  >
    <label
      className="col-sm-2 col-form-label"
      htmlFor="username"
    >
      Username:
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="is-valid form-control"
        id="username"
        name="username"
        placeholder="Enter your username"
        required={true}
        type="text"
      />
      <div
        className="invalid-feedback"
        invalid={true}
      >
        Username is incorrect
      </div>
    </div>
  </div>
  <div
    className="row form-group"
  >
    <label
      className="col-sm-2 col-form-label"
      htmlFor="password"
    >
      Password:
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="is-valid form-control"
        id="password"
        name="password"
        placeholder="Enter your password"
        required={true}
        type="password"
      />
      <div
        className="invalid-feedback"
        invalid={true}
      >
        Password is incorrect
      </div>
    </div>
  </div>
  <div
    className="text-center"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
  <br />
  <h1
    className="text-center text-success"
  >
    hello
     Logged In
  </h1>
</form>
`;
